lib/ansible/galaxy/collection/_types.py

```diff
@@ -0,0 +1,98 @@
+# Copyright: (c) 2025, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""A collection of shared types for ``ansible.galaxy.collection``."""
+
+from __future__ import annotations
+
+import queue as _q
+import typing as _t
+
+import ansible.module_utils.compat.typing as _tc
+from ansible.module_utils.common.sentinel import Sentinel as _Sentinel
+
+
+DisplayQueueItemType: _t.TypeAlias = tuple[
+    str,
+    tuple[_t.Any, ...],
+    dict[str, _t.Any],
+]
+DisplayQueueType: _t.TypeAlias = _q.Queue[DisplayQueueItemType]
+
+
+class DisplayThreadProto(_t.Protocol):
```

Comment:
does this make sense to have here or should it go to a more general _types  file?